pkg/certificate/providers/config.go

```diff
@@ -186,20 +186,20 @@ func (c *MRCProviderGenerator) getHashiVaultOSMCertificateManager(mrc *v1alpha2.
 	// A Vault address would have the following shape: "http://vault.default.svc.cluster.local:8200"
 	vaultAddr := fmt.Sprintf("%s://%s:%d", provider.Protocol, provider.Host, provider.Port)
 
-	// If the DefaultVaultToken is empty, query Vault token secret
-	var err error
-	vaultToken := c.DefaultVaultToken
-	if vaultToken == "" {
+	// If provider has secret ref filled, query Vault token secret
+	token := c.DefaultVaultToken
+	if provider.Token.SecretKeyRef.Name != "" && provider.Token.SecretKeyRef.Namespace != "" {
```

Comment:
Should we also check if the key != ""?
```suggestion
	if provider.Token.SecretKeyRef.Name != "" && provider.Token.SecretKeyRef.Namespace != "" && provider.Token.SecretKeyRef.Key != "" {
```

Reply:
is the key required?

Reply:
I believe so and I don't think we provide a default. I will confirm this.

Reply:
I think this would fail if the key wasn't set https://github.com/openservicemesh/osm/blob/0d36e289fcf3272e09d712286f5e21bde3ee60dd/pkg/certificate/providers/config.go#L219. I think I can confirm this by modifying the e2e tests or testing it manually.

Reply:
makes sense, I've added the check you suggested